pageserver/src/page_service.rs

```diff
@@ -649,16 +644,15 @@ impl PageServerHandler {
         pgb.flush().await?;
 
         /* Send a tarball of the latest layer on the timeline */
-        let mut writer = CopyDataSink {
-            pgb,
-            rt: tokio::runtime::Handle::current(),
-        };
-        tokio::task::block_in_place(|| {
+        {
+            let mut writer = pgb.copyout_writer();
             let basebackup =
                 basebackup::Basebackup::new(&mut writer, &timeline, lsn, prev_lsn, full_backup)?;
-            tracing::Span::current().record("lsn", basebackup.lsn.to_string().as_str());
-            basebackup.send_tarball()
-        })?;
+            // FIXME: instrument
+            //tracing::Span::current().record("lsn", basebackup.lsn.to_string().as_str());
```

Comment:
Not sure I follow this comment, we can do

`basebackup.send_tarball().instrument(info_span!("....")).await?;`

quite easily too.

Reply:
Note there's already a span for this method via `#[instrument]`. I had remembered that fields recorded later will need to be introduced as empty fields, but perhaps I remember wrong.

Reply:
Seems like you two talked past each other?
The point is that the commented-out code adds `lsn` to the current span, whereas @SomeoneToIgnore snippet creates a new span just for the send_tarball() call.

Let's stick to the current behavior and remove the comment. According to [the docs](https://docs.rs/tracing/latest/tracing/span/struct.Span.html#method.record) the way to do it is to use `field::Empty`.

I'm not sure about possible async footguns, but the docs don't highlight anything.

Reply:
Uncommented the original line. Not sure what the problem was that made me comment it out in the first place..